### PR TITLE
entities: sync for contributions and concepts

### PR DESCRIPTION
**Pull request by rerowep**
## Why are you opening this PR?

- Which task/US does it implement?
- Which issue does it fix?

## Dependencies

My PR depends on the following `rero-ils-ui`'s PR(s):

* rero/rero-ils-ui#<xx>

## How to test?

- What command should I have to run to test your PR?
- What should I test through the UI?

### CODE REVIEW

**Review by zannkukai — CHANGES_REQUESTED** [review]

File: rero_ils/modules/entities/sync.py [hunk 1]
```diff
@@ -38,8 +38,8 @@
     requests_retry_session, set_timestamp
 
 
-class SyncAgent(object):
-    """Agent MEF synchronization."""
+class SyncEntity(object):
```
[the remark this answers is not in the document]
Comment (rerowep) [reply rero_ils/modules/entities/sync.py:41]: Agent is unfortunately no longer the correct word and it was not be corrected beforehand.

File: rero_ils/config.py [hunk 2]
```diff
@@ -2978,6 +2978,11 @@ def _(x):
         ]
     }
 }
+RERO_ILS_ENTITY_TYPES = {
```
[the remark this answers is not in the document]
Comment (rerowep) [reply rero_ils/config.py:2981]: Ok I have done a cherry-pick with `release: v1.17.0`

**Review by zannkukai — APPROVED** [review]

File: rero_ils/modules/entities/sync.py [hunk 3]
```diff
@@ -325,48 +339,54 @@ def sync_record(self, pid):
...
             # get contribution in db
-            agent = Entity.get_record_by_pid(pid)
-            if not agent:
+            entity = Entity.get_record_by_pid(pid)
+            if not entity:
                 raise Exception(f'ERROR MEF {pid} does not exists in db.')
             self.logger.debug(f'Processing MEF(pid: {pid})')
-            # iterate over all agent sources: rero, gnd, idref
-            doc_pids = self._get_documents_pids_from_mef(agent.pid)
+            # iterate over all entity sources: rero, gnd, idref
```
Comment (PascalRepond) [inline rero_ils/modules/entities/sync.py:346]: Maybe for performance, if entity type is not bf:Person or bf:Organisation, you don't need to check gnd and rero, only idref? But I don't know if it's worth differentiating the behaviour.

**Review by PascalRepond — APPROVED** [review]

**Review by jma — APPROVED** [review]
bf:Topic sync tests seems missing.

**Pull request merged**